src/PDFWriterDriver.cpp

```diff
@@ -972,9 +984,21 @@ METHOD_RETURN_TYPE PDFWriterDriver::MergePDFPagesToPage(const ARGS_TYPE& args)
     if(caller.IsValid())
         pdfWriter->mPDFWriter.GetDocumentContext().AddDocumentContextExtender(&caller);
     
-    EStatusCode status = pdfWriter->mPDFWriter.MergePDFPagesToPage(page->GetPage(),
-                                                                    *String::Utf8Value(args[1]->ToString()),
-                                                                    pageRange);
+    EStatusCode status;
+    if(args[1]->IsString()) 
+    {
+        status = pdfWriter->mPDFWriter.MergePDFPagesToPage(page->GetPage(),
+                                                           *String::Utf8Value(args[1]->ToString()),
+                                                           pageRange);
+    }
+    else 
+    {
+        ObjectByteReaderWithPosition* proxy = new ObjectByteReaderWithPosition(args[1]->ToObject());
+        status = pdfWriter->mPDFWriter.MergePDFPagesToPage(page->GetPage(),

```

Comment:
its better to create a stack object. this way you don't forget to delete it when done. look at similar cases